text/0152-telemetry-schemas.md

```diff
@@ -0,0 +1,1160 @@
+URL. Language maintainers SHOULD choose the idiomatic way for their language.
+
+The effect of associating a Schema URL with a Tracer/Meter/LogEmitter SHOULD be
+that the schema_url in the InstrumentationLibrarySpans,
+InstrumentationLibraryMetrics, InstrumentationLibraryLogs message for all the
+telemetry emitted with the associated Tracer/Meter/LogEmitter is populated with
+the provided Schema URL value.
+
+If the Tracer/Meter/LogEmitter is not associated with a Schema URL then the
+exporter MUST leave the schema_url field in OTLP messages unset, in which case
+the application-wide Schema URL [will apply](#application-wide-schema-url).
+
+Open Question: how to make it easy for instrumentation libraries to refer to a
+particular OpenTelemetry schema version and also make sure any semantic
+convention helpers the library uses (e.g. constants that define the semantic
+conventions) match exactly that same schema version? One possible solution is to
+introduce helper packages per schema version that the libraries can use, e.g.
+constants that define the semantic conventions and the corresponding schema
+version url. This should be likely the topic for a follow-up OTEP.
+
+### Application-wide Schema URL
+
+The SDK interface MUST provide a way for the user to optionally set an
+application-wide Schema URL. This Schema URL will be populated in all
+ResourceSpans, ResourceMetrics and ResourceLogs messages emitted by OTLP
+Exporter.
+
+If the user does not set an application-wide Schema URL then the current Schema
+URL MUST be populated by OTLP Exporter in the messages, where "current" means
+the version of OpenTelemetry Schema against which the SDK is coded.
+
+Note that if there is a schema url associated with instrumentation library it
+overrides the application-wide schema url as described [here](#otlp-changes).
+
+## OpenTelemetry Schema 1.0.0
+
+The following is the full content of Schema File for OpenTelemetry Schema 1.0.0:
+
+```yaml
+file_format: 1.0.0
+schema_url: https://opentelemetry.io/schemas/1.0.0
+versions:
+  1.0.0:
+```
+
+Since this is the first version of OpenTelemetry schema there are no changes
+since the previous version (because there is no previous version) so the section
+1.0.0 is empty.
+
+This file SHOULD be available for retrieval at
+[https://opentelemetry.io/schemas/1.0.0](https://opentelemetry.io/schemas/1.0.0)
+
+All OpenTelemetry instrumentation solutions will follow this schema.
+
+Changes to OpenTelemetry semantic conventions SHOULD be handled as described in
+the [Schema Version Number](#schema-version-number) section. For example,
+addition of a completely new optional attribute to the semantic conventions
+SHOULD NOT require a new telemetry schema version. Similarly, a removal of an
+optional attribute SHOULD NOT require a new schema version, removal or addition
+of a required attribute to an existing entity SHOULD require a new schema
+version. Generally, any changes to the semantic conventions schema that do not
```

Comment:
If we add a new required attribute to the semantic convention, how will a new schema version file look like? It still will be empty, right? Because there is no transformation that changes previous version into a new one?

Reply:
Correct. The schema "changes" will be empty. However, I believe it is still valuable to record it as a new schema version, simply to indicate that it is different. The telemetry consumers are free to use or ignore that fact.
In the future we may add ability to explicitly record in the "changes" section the fact that a required attribute was added.